src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
You could have the dot_avx, etc implementations in a similar `#ifdef` guard for x86 architecture, and have them either return a bogus value on other architectures, or error at runtime somehow (`assert(false); return 0;`, so it doesn't crash in release), and let the dispatcher be responsible for not attempting to run implementations that won't work.